samples/nrf9160/modem_shell/src/shell.c

```diff
@@ -181,6 +181,23 @@ int lwm2m_carrier_event_handler(const lwm2m_carrier_event_t *event)
 }
 #endif
 
+int sleep_shell(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc == 2) {
+		long sleep_duration = strtol(argv[1], NULL, 10);
+
+		if (sleep_duration > 0) {
+			k_sleep(K_SECONDS(sleep_duration));
+			return 0;
+		}
+		mosh_print("sleep: duration must be 1...%ld\n", LONG_MAX);
+	} else {
+		mosh_print_no_format("Usage: sleep <time in seconds>\n");
+	}
+
+	return -EINVAL;
```

Comment:
If you change `SHELL_CMD_REGISTER` to `SHELL_CMD_ARG_REGISTER`:
```suggestion
	long sleep_duration = strtol(argv[1], NULL, 10);

	if (sleep_duration > 0) {
		k_sleep(K_SECONDS(sleep_duration));
		return 0;
	}
	mosh_print("sleep: duration must be 1...%ld\n", LONG_MAX);

	return -EINVAL;
```